### PR TITLE
fix (oom): added missing event fields

### PR DESCRIPTION
**Pull request by robinmacharg**
## Goal

To add fields that appear in non-OOM events to OOM events (codeBundleId, locale, osName and modelNumber). 

## Design

The fields are trivial additions that add values to the info cached by the OOM watchdog.

## Changeset

New fields added to the `generateCacheInfoWithConfig()` method of `BSGOutOfMemoryWatchdog`.

## Tests

Manual testing to ensure values were being sent through.  An additional unit test was added to ensure that expected keys/values are present in the cache.

### Outstanding Questions

There is some commonality with non-OOM fields.  Is there scope for generalising the event field population?

Expected fields are tested for.  There are no tests for unexpected fields; should there be?

- This pull request is ready for:
  - [ ] Initial review of the intended approach, not yet feature complete
  - [ ] Structural review of the classes, functions, and properties modified
  - [X] Final review

- [X] The correct target branch has been selected (`master` for fixes, `next` for
  features)
- [ ] Consistency across platforms for structures or concepts added or modified
- [X] Consistency between the changeset and the goal stated above
- [ ] Internal consistency with the rest of the library - is there any overlap between existing interfaces and any which have been added?
- [ ] Usage friction - is the proposed change in usage cumbersome or complicated?
- [ ] Performance and complexity - are there any cases of unexpected O(n^3) when iterating, recursing, flat mapping, etc?
- [ ] Concurrency concerns - if components are accessed asynchronously, what issues will arise
- [X] Thoroughness of added tests and any missing edge cases
- [ ] Idiomatic use of the language

### CODE REVIEW

File: Source/BSGOutOfMemoryWatchdog.m [hunk 1]
```diff
@@ -243,6 +243,8 @@ - (NSMutableDictionary *)generateCacheInfoWithConfig:(BugsnagConfiguration *)con
     app[@"releaseStage"] = config.releaseStage;
     app[@"version"] = systemInfo[@BSG_KSSystemField_BundleShortVersion] ?: @"";
     app[@"bundleVersion"] = systemInfo[@BSG_KSSystemField_BundleVersion] ?: @"";
+    // 'codeBundleId' only (optionally) exists for Unity clients and defaults otherwise to nil
```
Comment (kattrali) [inline Source/BSGOutOfMemoryWatchdog.m]:
```suggestion
    // 'codeBundleId' only (optionally) exists for React Native clients and defaults otherwise to nil
```
